modules/sovrnAnalyticsAdapter.js

```diff
@@ -0,0 +1,217 @@
+    this.body.prebidVersion = $$PREBID_GLOBAL$$.version
+    this.body.aid = affiliateId
+    this.body.winningBid = JSON.parse(JSON.stringify(event))
+    this.body.url = utils.getTopWindowLocation().href
+    this.body.payload = 'winner'
+    delete this.body.winningBid.ad
+  }
+
+  /**
+   * Sends the auction to the the ingest server
+   */
+  send() {
+    utils.logInfo(this.body)
+    ajax(
+      PBA_URL,
+      null,
+      JSON.stringify(this.body),
+      {
+        contentType: 'application/json',
+        method: 'POST',
+      }
+    )
+  }
+}
+
+/** Class representing an Auction */
+class AuctionData {
+  /**
+   * Create a new auction data collector
+   * @param {string} affiliateId - the affiliate id from the analytics config
+   * @param {string} auctionId - the auction id from the auction event
+   */
+  constructor(affiliateId, auctionId) {
+    this.auction = {}
+    this.auction.prebidVersion = $$PREBID_GLOBAL$$.version
+    this.auction.aid = affiliateId
+    this.auction.auctionId = auctionId
+    this.auction.payload = 'auction'
+    this.auction.timeouts = {
+      buffer: config.getConfig('timeoutBuffer'),
+      bidder: config.getConfig('bidderTimeout'),
+    }
+    this.auction.priceGranularity = config.getConfig('priceGranularity')
+    this.auction.url = utils.getTopWindowLocation().href
+    this.auction.requests = []
+    this.auction.unsynced = []
+    this.dropBidFields = ['auctionId', 'ad', 'requestId', 'bidderCode']
+  }
+
+  /**
+   * Record a bid request event
+   * @param {*} event - the args object from the auction event
+   */
+  bidRequested(event) {
+    delete event.doneCbCallCount
+    delete event.auctionId
+    this.auction.requests.push(JSON.parse(JSON.stringify(event)))
+  }
+
+  /**
+   * Finds the bid from the auction that the event is associated with
+   * @param {*} event - the args object from the auction event
+   * @return {*} - the bid
+   */
+  findBid(event) {
+    const bidder = this.auction.requests.find(r => (r.bidderCode === event.bidderCode))
+    if (!bidder) {
+      this.auction.unsynced.push(event)
+    }
+    let bid = bidder.bids.find(b => (b.bidId === event.requestId))
+
+    if (!bid) {
+      event.unmatched = true
+      bidder.bids.push(event)
+    }
+    return bid
+  }
+
+  /**
+   * Records the original bid before any adjustments have been made
+   * @param {*} event - the args object from the auction event
+   */
+  originalBid(event) {
+    let bid = this.findBid(event)
+    if (bid) {
+      Object.assign(bid, JSON.parse(JSON.stringify(event)))
+      this.dropBidFields.forEach((f) => delete bid[f])
+    }
+  }
+
+  /**
+   * Replaces original values with adjusted values and records the original values for changed values
+   * in bid.originalValues
+   * @param {*} event - the args object from the auction event
+   */
+  adjustedBid(event) {
+    let bid = this.findBid(event)
+    if (bid) {
+      bid.originalValues = Object.keys(event).reduce((o, k) => {
```

Comment:
I think you're right. I don't think the originalValues are getting populated correctly. Working on that now. (and there will be a test for it)

Reply:
False alarm. Jon pointed out that the bidadjustment occurs before the bidresponse. I added criteria to the bidadjustment test to verify the original values get populated. I also commented on this since its not obvious.